Assets/ShooterProject/Scripts/General/GameObjectsPool.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using UnityEngine;
+namespace ShooterProject.Scripts.General
+{
+	public class GameObjectsPool
+	{
+		#region Fields
+
+		private List<GameObject> _objects;
+
+		private bool _autoExpand;
+		private bool _activeByDefault;
+		private Transform _container;
+		private GameObject _objectPrefab;
+		private bool _initialized;
+
+		#endregion
+
+		#region Properties
+
+		public int MaxSize { get; }
+
+		#endregion
+
+		#region Constructors
+		/// <summary>
+		/// Конструктор класса
+		/// </summary>
+		/// <param name="maxSize">Максимальный размер пула</param>
+		/// <param name="autoExpand">Использовать ли авторасширение</param>
+		/// <param name="activeByDefault">Параметр active объектов при инициализации или добавлении в пул</param>
+		/// <param name="prefab">Префаб объекта</param>
+		/// <param name="container">Контейнер пула</param>
+		public GameObjectsPool(int maxSize, bool autoExpand, bool activeByDefault, GameObject prefab, Transform container)
+		{
+			MaxSize = maxSize;
+			_autoExpand = autoExpand;
+			_activeByDefault = activeByDefault;
+			_objectPrefab = prefab;
+			_container = container;
+
+			InitPool();
+		}
+
+		#endregion
+
+		#region Public Methods
+		/// <summary>
+		/// Проверяет наличие свободного элемента и возвращает его
+		/// </summary>
+		/// <param name="element">Свободный элемент пула</param>
+		/// <param name="allowLongTimeNotUsed">Если нет свободного элемента и нельзя создать новый, то возвращает давно не использовавшийся элемент</param>
+		/// <returns>Возвращает true если удалось найти свободный элемент, иначе возвращает false</returns>
+		public bool TryGetFreeElement(out GameObject element, bool allowLongTimeNotUsed)
+		{
+			if (!_initialized)
+				throw new System.Exception("Пул объектов не был инициализирован");
+			foreach (var poolObject in _objects)
+			{
+				if (!poolObject.gameObject.activeSelf)
```

Comment:
poolObject итак уже типа GameObject же, поэтому нет смысла получать его gameObject
```suggestion
				if (!poolObject.activeSelf)
```